src/action/list_directory.rs

```diff
@@ -169,81 +177,95 @@ fn get_accesses_time(metadata: &Metadata) -> Option<SystemTime> {
     }
 }
 
+/// Returns the last modification time of provided `Metadata`.
 fn get_modification_time(metadata: &Metadata) -> Option<SystemTime> {
     match metadata.modified() {
-        Ok(atime) => Some(atime),
+        Ok(mtime) => Some(mtime),
         Err(err) => {
             warn!("unable to get last modification time: {}", err);
             None
         }
     }
 }
 
+/// Returns the creation time of provided `Metadata`.
 fn get_creation_time(metadata: &Metadata) -> Option<SystemTime> {
     match metadata.created() {
-        Ok(atime) => Some(atime),
+        Ok(creation_time) => Some(creation_time),
         Err(err) => {
             warn!("unable to get creation time: {}", err);
             None
         }
     }
 }
 
+/// Returns the last status change time of provided `Metadata`.
 #[cfg(target_os = "linux")]
 fn get_status_change_time(metadata: &Metadata) -> Option<SystemTime> {
-    use std::time::Duration;
+    use std::time::{Duration, UNIX_EPOCH};
     use std::os::unix::fs::MetadataExt;
 
     UNIX_EPOCH.checked_add(Duration::from_secs(metadata.ctime() as u64))
 }
 
+/// Reads a symbolic link, returning the path to the file that the link points to.
+#[cfg(target_os = "linux")]
+fn get_symlink(metadata: &Metadata, file_path: &Path) -> Option<PathBuf> {
+    if metadata.file_type().is_symlink() {
```

Comment:
Prefer to use early return to avoid nesting.

Reply:
Fixed.